lib/Runtime/Language/SourceTextModuleRecord.cpp

```diff
@@ -373,7 +393,16 @@ namespace Js
             });
             if (localNameId != Js::Constants::NoProperty)
             {
-                resolvedExportMap->AddNew(exportName, { this, localNameId });
+                ModuleRecordBase* sourceModule = this;
+                ModuleNameRecord* importRecord = nullptr;
+                if (this->importRecordList != nullptr
+                    && this->ResolveImport(localNameId, &importRecord) 
```

Comment:
This is checking for the case of exporting a binding we imported but didn't use the re-export syntax. It would look like:
import { foo } from 'module';
export { foo };

I'll add a comment, it's a little confusing.